plugins/org.jkiss.dbeaver.ext.postgresql/src/org/jkiss/dbeaver/ext/postgresql/model/impls/redshift/RedshiftView.java

```diff
@@ -98,43 +99,34 @@ private boolean isViewVwithNoSchemaBinding(DBRProgressMonitor monitor) throws DB
                         String colType = JDBCUtils.safeGetString(dbResult, "col_type");
                         int colNum = JDBCUtils.safeGetInt(dbResult, "col_num");
 
-                        PostgreTableColumn viewColumn = new PostgreTableColumn(this);
-                        viewColumn.setName(colName);
-                        viewColumn.setPersisted(true);
-
-                        String colTypeName;
-                        int colPrecision = -1, colScale = -1;
-                        int divPos = colType.indexOf("(");
-                        if (divPos != -1) {
-                            colTypeName = colType.substring(0, divPos);
-                            int divPos2 = colType.lastIndexOf(")");
-                            if (divPos2 < 0) {
-                                log.error("Bad column type name '" + colType + "'");
-                                continue;
-                            }
-                            String typeMod = colType.substring(divPos + 1, divPos2).trim();
-                            divPos2 = typeMod.indexOf(",");
-                            if (divPos2 < 0) {
-                                colPrecision = CommonUtils.toInt(typeMod);
-                            } else {
-                                colPrecision = CommonUtils.toInt(typeMod.substring(0, divPos2).trim());
-                                colScale = CommonUtils.toInt(typeMod.substring(divPos2 + 1).trim());
-                            }
-                        } else {
-                            colTypeName = colType;
-                        }
+			String colTypeName;
+			int divPos = colType.indexOf("(");
+			if (divPos != -1) {
+			    colTypeName = colType.substring(0, divPos);
+			    int divPos2 = colType.lastIndexOf(")");
+			    if (divPos2 < 0) {
+				log.error("Bad column type name '" + colType + "'");
+				continue;
+			    }
+			} else {
+			    colTypeName = colType;
+			}
```

Comment:
Instead of manually doing this, I suggest you to use `DBUtils#getTypeModifiers` instead. See its javadoc for usage

Reply:
That piece of code was already there in that class. But will refactor. Thanks for the suggestion.

Reply:
Completed that change.